pkg/oam/util/containerized/helper.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package containerized
+
+import (
+	"context"
+	"reflect"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+var (
+	serviceKind          = reflect.TypeOf(corev1.Service{}).Name()
+	serviceAPIVersion    = corev1.SchemeGroupVersion.String()
+	deploymentKind       = reflect.TypeOf(appsv1.Deployment{}).Name()
+	deploymentAPIVersion = appsv1.SchemeGroupVersion.String()
+)
+
+// Reconcile error strings.
+const (
+	defaultNamespace = "default"
+
+	labelKey = "containerizedworkload.oam.crossplane.io"
+)
+
+// ServiceInjector adds a Service object for the first Port on the first
+// Container for the first Deployment observed in a workload translation.
+func ServiceInjector(ctx context.Context, w metav1.Object, objs []metav1.Object) ([]metav1.Object, error) {
```

Comment:
Looks like this got lost when things were moved from `crossplane` to `oam-kubernetes-runtime`. I'll take care of getting it added to `oam-kubernetes-runtime` so that it doesn't have to be re-implemented multiple times :) https://github.com/crossplane/crossplane/blob/8f7ffc42936f8ae96612e82bae5e1437065bdbe5/pkg/oam/workload/translate.go#L101

Reply:
sure, I will remove this part when this reappears in the oam-k8s-runtime repo.

Reply:
cc @ryanzhang-oss https://github.com/crossplane/oam-kubernetes-runtime/pull/21